kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/VariableDeclaration.java

```diff
@@ -56,9 +56,7 @@ public static VariableDeclaration fromString(String encoded) {
         String[] split = encoded.split(":");
         String identifier = split[0];
         String type = (split.length == 2 || split.length == 3) ? split[1] : "";
-        if (identifier.isEmpty()) {
```

Comment:
This means that now, in case the identifier of the variables comes empty, we'll be creating anyway the variable declaration in our model? Just asking, not sure if makes sense creating it or skipping... maybe if identifier is empty we can check the value, and if empty as well, then just not even create the variable declaration in our model, otherwise doe sin the UI will appear an "empty row"?

Reply:
I tried creating some process variables with empty name and they were not added to the BPMN file. Was I checking it wrong?

Reply:
Is this class only used for Multiple instance subprocess?
Seems to be working well:
```
  <bpmn2:itemDefinition id="_9337DB9F-80F8-4E3B-B145-DF3ABA77A38D_multiInstanceItemType_inputDataMI" structureRef="Float"/>
  <bpmn2:itemDefinition id="_9337DB9F-80F8-4E3B-B145-DF3ABA77A38D_multiInstanceItemType_" structureRef="Float"/>
  ```

Reply:
So it seems that now we can also import processes with these kind of entries (UI does not allow this to be done)
Globals
```
      <drools:import name="ok"/>
      <drools:import/>
      <drools:global identifier="" type="assssss"/>
      <drools:global identifier="" type="Object"/>
  ```
  Editor will not show these.
  Process Vars
  ```
      <bpmn2:property id="" itemSubjectRef="_aItem" name=""/>
 ```

Reply:
Seems like desired behavior, could you please confirm?

Reply:
> This means that now, in case the identifier of the variables comes empty, we'll be creating anyway the variable declaration in our model? Just asking, not sure if makes sense creating it or skipping... maybe if identifier is empty we can check the value, and if empty as well, then just not even create the variable declaration in our model, otherwise doe sin the UI will appear an "empty row"?

Hi @romartin basically if it has no name, it will just be the type, since it uses format variable_name_variable_type

Reply:
So when this is in UI:
![Screenshot from 2021-07-16 08-49-07](https://user-images.githubusercontent.com/5937251/125904638-6f4b2eeb-ece6-441f-998a-1f2b96eac4c2.png)

XML contains:
```
 <bpmn2:itemDefinition id="_9337DB9F-80F8-4E3B-B145-DF3ABA77A38D_multiInstanceItemType_inputDataMI" structureRef="Float"/>
  <bpmn2:itemDefinition id="_9337DB9F-80F8-4E3B-B145-DF3ABA77A38D_multiInstanceItemType_" structureRef="Float"/>
```
I understand this is the expected fix, could you please confirm?

Reply:
Hi @domhanak yes, that is correct